### PR TITLE
[STY] Fix PTH123

### PR DESCRIPTION
**Pull request by hndgzkn**
- Closes  none but relates to #4327 

Changes proposed in this pull request:

fixes error PTH 123: https://docs.astral.sh/ruff/rules/builtin-open/

except for `nilearn/datasets` package.

### CODE REVIEW

**Comment by hndgzkn** [comment]
All appearences of PTH123 apart from `nilearn/datasets` package are fixed. There remains 39 errors in `nilearn/datasets`. We can review this part and merge it. How does it sound @Remi-Gau ?

**Comment by Remi-Gau** [comment]
> All appearences of PTH123 apart from `nilearn/datasets` package are fixed. There remains 39 errors in `nilearn/datasets`. We can review current changes and merge it, and do dataset part with another PR. How does it sound @Remi-Gau ?

good with me: we can silence the errors in dataset via the pyproject.toml

File: nilearn/_utils/numpy_conversions.py [hunk 1]
```diff
@@ -3,11 +3,10 @@
...
 import csv
+from pathlib import Path
 
 import numpy as np
 
-from .helpers import stringify_path
-
 
 def _asarray(arr, dtype=None, order=None):
     # np.asarray does not take "K" and "A" orders in version 1.3.0
```
Comment (Remi-Gau) [inline nilearn/_utils/numpy_conversions.py:12]:
```suggestion
    # np.asarray does not take "K" and "A" orders in version 1.3.0
```

unrelated: seeing makes me think that we can probably simplify this function as it seems to handle behavior for old versions of numpy we do not support anymore

probably worth checking in another issue / PR

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
@hndgzkn ping me if you need another review

**Comment by hndgzkn** [comment]
> @hndgzkn ping me if you need another review

I've silenced the error for datasets package, no other changes. Needs a review from another person.

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by Remi-Gau** [comment]
waiting for dov build to finish

**Pull request merged**